controllers/btpoperator_controller.go

```diff
@@ -520,22 +536,150 @@ func (r *BtpOperatorReconciler) reconcileResources(ctx context.Context, s *corev
 	return nil
 }
 
-func (r *BtpOperatorReconciler) getResourcesToApplyPath() string {
-	return fmt.Sprintf("%s%capply", ResourcesPath, os.PathSeparator)
+func (r *BtpOperatorReconciler) handleSecretChanges(ctx context.Context, resourcesToApply []*unstructured.Unstructured, managerSecret *corev1.Secret, logger *logr.Logger) error {
+	// this is first scenario when provisoning. We cannot restart operator deployment here since resources are not yet applied
+	operator := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      DeploymentName,
+			Namespace: ChartNamespace,
+		},
+	}
+	if err := r.Get(ctx, client.ObjectKeyFromObject(operator), operator); err != nil {
+		if k8serrors.IsNotFound(err) {
+			logger.Info("operator deployment not found")
+			return nil
+		}
+		logger.Error(err, "while getting operator deployment")
+		return fmt.Errorf("while operator deployment: %w", err)
+	}
+
+	configMapIndex, secretIndex := r.GetConfigMapAndSecretIndexes(resourcesToApply)
+	configMap := resourcesToApply[configMapIndex]
+	operatorSecret := resourcesToApply[secretIndex]
+
+	operatorNeedRestart, err := r.verifySecretChanges(ctx, configMap, operatorSecret, managerSecret, logger)
+	if err != nil {
+		return fmt.Errorf("failed to reconcile manager secret changes: %w", err)
+	}
+	if operatorNeedRestart {
```

Comment:
Still we do not handle State changes (should be in Processing state during operator restart).